src/home_robot/home_robot/mapping/voxel.py

```diff
@@ -23,3 +26,49 @@ def combine_point_clouds(
         pc_xyz = np.concatenate([pc_xyz, xyz], axis=0)
     pcd = numpy_to_pcd(pc_xyz, pc_rgb).voxel_down_sample(voxel_size=sparse_voxel_size)
     return pcd_to_numpy(pcd)
+
+
+class SparseVoxelMap(object):
+    """Create a voxel map object which captures 3d information."""
+
+    def __init__(self, resolution=0.01, feature_dim=3):
+        self.resolution = resolution
+        self.feature_dim = feature_dim
+        self.xyz = None
+        self.feats = None
+        self.observations = []
+
+    def add(self, camera_pose: np.ndarray, xyz: np.ndarray, feats: np.ndarray):
+        """Add this to our history of observations. Also update the current running map."""
+        assert xyz.shape[-1] == 3
+        assert feats.shape[-1] == self.feature_dim
+        assert xyz.shape[0] == feats.shape[0]
+        self.observations.append((camera_pose, xyz, feats))
+        world_xyz = trimesh.transform_points(xyz, camera_pose)
+
+        if self.xyz is None:
+            self.xyz = world_xyz
+            self.feats = feats
```

Comment:
there should be an else here right? Voxelization will get rid of duplicate points, but better to be explicit.

Reply:
Or send in `self.xyz=None, self.feats=None` into `combine_point_clouds` as is and let `combine_point_clouds` handle None checking and skipping unnecessary combination.

Reply:
yeah this was a bug, just added the else